src/commands/init/index.ts

```diff
@@ -79,30 +142,56 @@ export default class Init extends Command {
         return projectConfig;
     }
 
-    async ensurePackagesAreDownloaded(projectConfig: ProjectConfig, force: boolean, verbose: boolean) {
-        for (const packageConfig of projectConfig.getPackages()) {
-            const packageVersions = await packageConfig.getPackageVersions();
-            const packageVersion = resolveVersionIdentifier(packageVersions, packageConfig.version);
+    private async _resolveVersion(packageConfig: PackageConfig, verbose: boolean) {
+        const packageVersions = await packageConfig.getPackageVersions();
+        const packageVersion = resolveVersionIdentifier(packageVersions, packageConfig.version);
 
-            if (verbose) {
-                this.log(`... Resolved '${packageConfig.getPackageName()}:${packageConfig.version}' to version: '${packageVersion}'`);
-            }
+        if (verbose) {
+            this.log(`... Resolved '${packageConfig.getPackageName()}:${packageConfig.version}' to version: '${packageVersion}'`);
+        }
 
-            packageConfig.setPackageVersion(packageVersion);
+        packageConfig.setPackageVersion(packageVersion);
+    }
 
-            if (!force && packageConfig.isPackageInstalled()) {
-                this.log(`... '${packageConfig.getPackageName()}:${packageConfig.version}' already installed.`);
-                continue;
+    private async _modifyProjectConfig(
+        requestedPackageConfig: PackageConfig,
+        projectConfig: ProjectConfig,
+        force: boolean,
+        verbose: boolean,
+    ) {
+        const packageConfigs = projectConfig
+            .getPackages()
+            .filter((config) => config.getPackageName() === requestedPackageConfig.getPackageName());
+
+        const isPackageExisting = packageConfigs.length >= 1;
+        if (!isPackageExisting) {
+            this.log(`... Package '${requestedPackageConfig.getPackageName()}:${requestedPackageConfig.version}' added to .velocitas.json`);
+            projectConfig.addPackage(requestedPackageConfig);
+
```

Comment:
the logic of adding a new package and checking if already existing should be done in project-config.ts "addPackage"
_modifyProjectConfig can than be shortened and be part of "_handleSinglePackageInit"

Reply:
```

addPackage(packageConfig: PackageConfig): boolean {
    const existingPackage = this._packages.find(config => config.getPackageName() === packageConfig.getPackageName());

    if (existingPackage) {
        if (existingPackage.version !== packageConfig.version) {
            existingPackage.setPackageVersion(packageConfig.version);
            return true;
        }
    } else {
        this._packages.push(packageConfig);
        return true;
    }
    return false;
}
```
Than you can get rid of _modifyProjectConfig completely and put this

```
        const isAdded = projectConfig.addPackage(requestedPackageConfig);

        if (isAdded) {
            this.log(`... Adding '${requestedPackageConfig.getPackageName()}:${requestedPackageConfig.version}' to .velocitas.json`);
        }
```
into "_handleSinglePackageInit"

Reply:
Thanks for your sophisticated answer. 

I will go for the add / get / update solution due to clearer single responsibilities of the methods.